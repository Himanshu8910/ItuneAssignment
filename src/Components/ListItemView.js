import React, { memo } from 'react';
import { Image, StyleSheet, Text, View, Dimensions } from 'react-native';
import { dateFormat } from '../Helpers/DateFormatter'
let deviceWidth = Dimensions.get('window').width

const ListItemView = ({ data, style, imageStyle }) => {
    const {
        artistName,
        artworkUrl100,
        collectionName,
        releaseDate,
    } = data;

    return (
        <View style={[styles.container, style]}>
            <View>
                <Image
                    style={[styles.imageViewStyle, imageStyle]}
                    source={{ uri: artworkUrl100 }}
                />
                <View style={styles.textContainerViewStyle}>
                    <Text style={styles.artistNameStyle} numberOfLines={1}>
                        {artistName}
                    </Text>
                    <View style={{ marginTop: 8 }}>
                        <Text style={styles.textColorGray} numberOfLines={1}>
                            Release date: <Text style={styles.textColorWhite}>{dateFormat(new Date(releaseDate))}</Text>
                        </Text>

                        <Text style={styles.textColorGray} numberOfLines={2}>
                            {/* {releaseDate} */}
                            Collection: <Text style={styles.textColorWhite}>{collectionName}</Text>
                        </Text>
                    </View>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        height: 240,
        margin: 4,
        backgroundColor: 'rgb(46,46,46)',
        width: (deviceWidth - 16) / 2,
        borderWidth: 1,
        borderColor: 'black'
    },
    imageViewStyle: {
        width: (deviceWidth - 26) / 2,
        height: 150,
        marginRight: 10,
    },
    details: {
        justifyContent: 'space-between',
    },
    moreDetails: {
        marginTop: 20,
    },
    description: {
        fontSize: 14,
        color: 'white',
    },
    textContainerViewStyle: {
        margin: 4
    },
    artistNameStyle: {
        fontSize: 16,
        color: 'white',
        fontWeight: 'bold'
    },
    textColorWhite: {
        color: 'white'
    },
    textColorGray: {
        color: '#999999'
    }
});

export default memo(ListItemView);