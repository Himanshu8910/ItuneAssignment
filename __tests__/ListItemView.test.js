import React from 'react';
import renderer from 'react-test-renderer';

import ListItemView from '../src/Components/ListItemView';
import mockData from './mockData.json';

const item = mockData.results[0];

test('renders correctly with data', () => {
  const tree = renderer.create(<ListItemView data={item} />).toJSON();
  expect(tree).toMatchSnapshot();
});

