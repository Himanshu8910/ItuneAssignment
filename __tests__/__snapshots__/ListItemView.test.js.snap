// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "rgb(46,46,46)",
        "borderColor": "black",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 240,
        "margin": 4,
        "width": 367,
      },
      undefined,
    ]
  }
>
  <View>
    <Image
      source={
        Object {
          "uri": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
        }
      }
      style={
        Array [
          Object {
            "height": 150,
            "marginRight": 10,
            "width": 362,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "margin": 4,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Jack Johnson
      </Text>
      <View
        style={
          Object {
            "marginTop": 8,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          Release date: 
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            01-Mar-2005
          </Text>
        </Text>
        <Text
          numberOfLines={2}
          style={
            Object {
              "color": "#999999",
            }
          }
        >
          Collection: 
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            In Between Dreams (Bonus Track Version)
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
